akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpResponseRendererFactory.scala

```diff
@@ -278,6 +242,90 @@ private[http] class HttpResponseRendererFactory(
 
 /**
  * INTERNAL API
+ *
+ * Shared base logic for rendering response headers with common logic for http 1 and 2
+ *
+ * @tparam T a mutable builder of some kind to build up the headers in
+ */
+private[http] abstract class AbstractHeaderRenderer[T] {
```

Comment:
Was hoping I could make it stateless rather than create a new one for every response/stage, but not sure that is possible